server/src/main/java/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -2350,7 +2350,7 @@ public boolean finalizeCommandsOnStart(final Commands cmds, final VirtualMachine
             if (reprogramGuestNtwks) {
                 finalizeIpAssocForNetwork(cmds, router, provider, guestNetworkId, null);
                 finalizeNetworkRulesForNetwork(cmds, router, provider, guestNetworkId);
-                finalizeMonitorService(cmds, profile, router, provider, guestNetworkId, true);
+                finalizeMonitorService(cmds, profile, router, provider, guestNetworkId, true, getRouterHealthChecksConfig(router));
```

Comment:
```suggestion
                finalizeMonitorService(cmds, profile, router, provider, guestNetworkId, true, routerHealthCheckConfig);
```

@phsm seems it is called in loop here, so better get the map outside loop and pass here.

Reply:
Good catch, updated

Reply:
> Good catch, updated

thanks @phsm